src/ui/components/search/searchcomponent.js

```diff
@@ -344,59 +374,96 @@ export default class SearchComponent extends Component {
     }
   }
 
-  requestIconAnimationFrame (iconState) {
+  requestIconAnimationFrame (iconState, iconElClass) {
     if (this.iconState === iconState) {
       return;
     }
+    const prevState = this.iconState;
     this.iconState = iconState;
     if (!this.isRequestingAnimationFrame) {
       this.isRequestingAnimationFrame = true;
       window.requestAnimationFrame(() => {
-        this.forwardIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
-        this.reverseIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
+        const iconEls = DOM.queryAll('.js-yxt-SearchBar-Icon');
```

Comment:
Is it possible to set the parent to this container like this: `DOM.queryAll(this._container, '.js-yxt-SearchBar-Icon');`,  or do we have to search the whole document?

Reply:
good point, just this container should be fine.